tests/templates/test_subroutines/test_qsvt.py

```diff
@@ -792,3 +792,68 @@ def test_raise_error(self, poly, routine, angle_solver, msg_match):
 
         with pytest.raises(AssertionError, match=msg_match):
             _ = qml.poly_to_angles(poly, routine, angle_solver)
+
+    def test_immutable_input(self):
+        """Test `poly_to_angles` does not modify the input"""
+
+        poly = [0, 1.0, 0, -1 / 2, 0, 1 / 3, 0]
+        poly_copy = poly.copy()
+        qml.poly_to_angles(poly, "QSVT")
+
+        assert len(poly) == len(poly_copy)
+
+        for p, q in zip(poly, poly_copy):
+            assert np.isclose(p, q)
```

Comment:
I think this should be possible?
```suggestion
        assert len(poly) == len(poly_copy)
        assert np.allclose(poly, poly_copy)
```